### PR TITLE
feat: add `addChain` method for dapps

### LINKED ISSUE #623: add addChain method using @argent/get-starknet
metamask method like this:
```
window.ethereum.request({
        method: 'wallet_addEthereumChain',
        params: [{
            chainId: string; // A 0x-prefixed hexadecimal string
            chainName: string;
            nativeCurrency: {
                name: string;
                symbol: string; // 2-6 characters long
                decimals: 18;
            };
            rpcUrls: string[];
            blockExplorerUrls?: string[];
            iconUrls?: string[]; // Currently ignored.
        }]
    })
    .catch((error) => {
        console.log(error)
    })
``` 

https://docs.metamask.io/guide/rpc-api.html#unrestricted-methods

### PR DESCRIPTION
**Pull request by dhruvkelawala**
closes #623

### CODE REVIEW

File: packages/extension/src/inpage.ts [hunk 1]
```diff
@@ -105,6 +109,62 @@ const starknetWindowObject: StarknetWindowObject = {
...
+          }),
+      ])
+
+      if (result === "error") {
+        throw Error("User abort")
+      }
+      if (result === "timeout") {
+        throw Error("User action timed out")
+      }
+
```
Comment (delaaxe) [inline packages/extension/src/inpage.ts:167]: Now that this `StarknetWindowObject.request` method is becoming pretty fat, can we factor out 2 functions, one for each of the `wallet_watchAsset` & `wallet_addStarknetChain` cases?

File: packages/extension/src/ui/features/actions/ActionScreen.tsx [hunk 2]
```diff
@@ -61,6 +62,27 @@ export const ActionScreen: FC = () => {
         />
       )
 
+    case "REQUEST_CUSTOM_NETWORK":
+      // action.payload
```
Comment (delaaxe) [inline packages/extension/src/ui/features/actions/ActionScreen.tsx:66]: commented code

File: packages/extension/src/ui/features/actions/AddNetworkScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  //   const compiledData = {
+  //     address: networkChainId,
+  //     ...(tokenDetails ?? {}),
+  //     ...(!tokenDetails?.name && { name: networkName }),
+  //     ...(!tokenDetails?.symbol && { symbol: networkBaseUrl }),
+  //     ...(!tokenDetails?.decimals && {
+  //       decimals: BigNumber.from(networkExplorerUrl || "0"),
+  //     }),
+  //     networkId: switcherNetworkId,
+  //   }
```
Comment (delaaxe) [inline packages/extension/src/ui/features/actions/AddNetworkScreen.tsx:97]: commented code

File: packages/extension/src/ui/features/actions/AddNetworkScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    <>
+      <Header hide={hideBackButton}>
+        <BackButton />
+      </Header>
+
+      <AddTokenScreenWrapper>
+        <H2>Add Network</H2>
+
+        <form
+          onSubmit={(e: React.FormEvent) => {
```
Comment (delaaxe) [inline packages/extension/src/ui/features/actions/AddNetworkScreen.tsx:109]: type is inferred here

```suggestion
          onSubmit={(e) => {
```

File: packages/extension/src/ui/features/actions/AddNetworkScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+            e.preventDefault()
+            if (requestedNetwork) {
+              try {
+                const networks = addNetworks(requestedNetwork)
+                console.log(
+                  "🚀 ~ file: AddNetworkScreen.tsx ~ line 139 ~ networks",
```
Comment (delaaxe) [inline packages/extension/src/ui/features/actions/AddNetworkScreen.tsx:115]: is there a tool that generates this?

File: packages/extension/src/ui/features/actions/AddNetworkScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+                  networks,
+                )
+                onSubmit?.()
+                navigate(routes.settingsNetworks())
+              } catch (e) {
```
Comment (delaaxe) [inline packages/extension/src/ui/features/actions/AddNetworkScreen.tsx:120]:
```suggestion
              } catch {
```

**Review by delaaxe — COMMENTED** [review]
good job

File: packages/extension/src/inpage.ts (hunk 1 above)
Comment (dhruvkelawala) [reply packages/extension/src/inpage.ts:167]: @janek26 and I had a call yesterday and felt that many files need refactoring. Should we refactor this function along with that or should I do it in this PR?

Comment (delaaxe) [reply packages/extension/src/inpage.ts:167]: it's ok to do some refactoring in feature PRs to keep things easy to understand, but yes bigger refactors should be in their own PRs

**Review by delaaxe — CHANGES_REQUESTED** [review]
much better now, just make sure `yarn build` still passes

**Comment by dhruvkelawala** [comment]
@delaaxe This should fix build

**Pull request merged**